### ISSUE TITLE
Map O,o and code editing commands (commenting, indenting, etc...) to vscode commands

### ISSUE THREAD

**Issue by asvetliakov** [asvetliakov/vscode-neovim#4]
- [x] o, O
- [x] formatting (= / ==)
- [x] vim-commentary like gc/gcc
- [x] set vim buffer tab options from vscode to allow ```>>``` and ```<<``` work
- [ ] folding

**Comment by asvetliakov** [comment]
Folding has separate issue #58

**Comment by pablopunk** [comment]
@asvetliakov do '>>' and '<<' work? They don't work for me

**Comment by asvetliakov** [comment]
@pablopunk What is your vscode indent options? tab indentation is not working currently, see #81 . Spaces should be fine

**Comment by pablopunk** [comment]
@asvetliakov I do have spaces. But it's weird, I just realized I can `s-tab` to indent left but `tab` won't do anything. Also, still no luck with `>>` and `<<`.

**Comment by asvetliakov** [comment]
@pablopunk ```tab``` is to navigate forward in jumplist in vim. ```s-tab``` mapping is coming from vscode. what is your space size?

**Comment by pablopunk** [comment]
2 spaces, but with 4 it doesn't work either

**Comment by asvetliakov** [comment]
curious, it should work with 4 spaces at least... do you have any tab/space related options (```expandtab/shiftwidth/etc...```) in your init.vim? does ```:retab``` command helps?

**Comment by pablopunk** [comment]
@asvetliakov with an empty init.vim it does work, I'll investigate what option is the one causing the issue 👍

**Comment by pablopunk** [comment]
Okay, my bad... I had these on my init.vim:

```viml
nnoremap < <Nop>
nnoremap < <Nop>
nnoremap < <Nop>
```

**Comment by asvetliakov** [comment]
Great to hear! feel free to reopen if it doesn't work correctly with your vscode's indent settings

**Comment by pablopunk** [comment]
Okay my bad... I had this on my config:

```
nnoremap < <Nop>
vnoremap < <Nop>
nnoremap > <Nop>
vnoremap > <Nop>
```

The reason is that I have maps for tab to work as `>>`:

```
nnoremap <tab> >>
nnoremap <s-tab> <<
```

But it seems that doesn't work inside vscode 😓

**Comment by asvetliakov** [comment]
```<s-tab>``` is not being sent to neovim because it has vscode mapping, try to put into your ```keybindings.json```:
```json
{
  "key": "shift+tab",
  "command": "-outdent"
}
```

this should unblock ```<s-tab>``` but may be additional work is needed to pass it to neovim.
However ```<tab>``` should work in your case

**Comment by pablopunk** [comment]
the opposite, actually. s-tab works but tab does not

**Comment by asvetliakov** [comment]
it works because it's bound to vscode action 😄 anyway, i created #92 to investigate/fix your keybindings

**Comment by pablopunk** [comment]
@asvetliakov great, thank you!
